backends/gdx-backends-gwt/src/com/google/gwt/webgl/client/WebGL2RenderingContext.java

```diff
@@ -0,0 +1,555 @@
+/*******************************************************************************
+ * Copyright 2022 See AUTHORS file.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.google.gwt.webgl.client;
+
+import com.google.gwt.canvas.dom.client.ImageData;
+import com.google.gwt.core.client.JavaScriptObject;
+import com.google.gwt.core.client.JsArrayBoolean;
+import com.google.gwt.core.client.JsArrayInteger;
+import com.google.gwt.core.client.JsArrayString;
+import com.google.gwt.dom.client.CanvasElement;
+import com.google.gwt.dom.client.ImageElement;
+import com.google.gwt.dom.client.VideoElement;
+import com.google.gwt.typedarrays.shared.ArrayBufferView;
+import com.google.gwt.typedarrays.shared.Float32Array;
+import com.google.gwt.typedarrays.shared.Int32Array;
+import com.google.gwt.typedarrays.shared.Uint32Array;
+
+/** @author Simon Gerst
+ * @author JamesTKhan */
+public class WebGL2RenderingContext extends WebGLRenderingContext {
+
+	/** Returns a WebGL2 context for the given canvas element. Returns null if no 3d context is available. */
+	public static WebGL2RenderingContext getContext (CanvasElement canvas) {
+		return getContext(canvas, WebGLContextAttributes.create());
+	}
+
+	/** Returns a WebGL2 context for the given canvas element. Returns null if no 3d context is available. */
+	public static native WebGL2RenderingContext getContext (CanvasElement canvas, WebGLContextAttributes attributes) /*-{
+		try {
+			var ctx = canvas.getContext("webgl2", attributes);
+			if (ctx != null) {
+				console.log('WebGL2 Enabled');
+				if ($wnd.WebGLDebugUtils) {
+					if ($wnd.console && $wnd.console.log) {
+						console.log('WebGL debugging enabled');
+					}
+					return $wnd.WebGLDebugUtils.makeDebugContext(ctx);
+				}
+				return ctx;
+			}
+		} catch (e) {
+			console.log(e);
+		}
+		return null;
+	}-*/;
+
+	protected WebGL2RenderingContext () {
+	}
+
+	public static JsArrayString toJsArray (String[] input) {
```

Comment:
private?

Reply:
Maybe even put it here https://github.com/libgdx/libgdx/blob/master/backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/GwtUtils.java

Reply:
I liked the idea of putting it in GwtUtils but it looks like it would not be available unless I update inherited GWT modules. For now will mark it private unless we want to move it into GwtUtils and update module inheritance.

Reply:
I don't see a reason why it wouldn't be available... But I don't really care

Reply:
Maybe it is a false positive in the IDE. This is warning I get but when I just tried moving it over I was able to still start up tests without updating the modules.

![image](https://user-images.githubusercontent.com/28971753/213735673-24349daa-c389-4f13-a946-f2b79dec03b0.png)

Reply:
Ah, I didn't see that it's in the Google package. Good possible that your warning is true then